adapters/IronSource/IronSourceAdapter/GADMAdapterIronSourceInterstitialAd.m

```diff
@@ -0,0 +1,174 @@
+// The class-level instance mapping from instance id to object reference
+static NSMapTable<NSString *, GADMAdapterIronSourceInterstitialAd *> *interstitialAdapterDelegates = nil;
+// The class-level delegate handling callbacks for all instances
+static GADMAdapterIronSourceInterstitialAdDelegate *interstitialDelegate = nil;
+
++ (void)initialize {
+    interstitialAdapterDelegates = [NSMapTable mapTableWithKeyOptions:NSPointerFunctionsCopyIn valueOptions:NSPointerFunctionsWeakMemory];
+    interstitialDelegate = [[GADMAdapterIronSourceInterstitialAdDelegate alloc] init];
+}
+
+#pragma mark - Load functionality
+
+- (void)loadInterstitialForAdConfiguration:(GADMediationInterstitialAdConfiguration*)adConfiguration
+                         completionHandler:(GADMediationInterstitialLoadCompletionHandler)
+completionHandler {
+    
+    _interstitalAdLoadCompletionHandler = completionHandler;
+    // Default instance state
+    self.instanceState = GADMAdapterIronSourceInstanceStateStart;
+    NSDictionary *credentials = [adConfiguration.credentials settings];
+    
+    /* Parse application key */
+    NSString *applicationKey = credentials[GADMAdapterIronSourceAppKey];
+    if (applicationKey != nil && ![GADMAdapterIronSourceUtils isEmpty:applicationKey]) {
+        applicationKey = credentials[GADMAdapterIronSourceAppKey];
+    } else {
+        NSError *error = GADMAdapterIronSourceErrorWithCodeAndDescription(
+                                                                          GADMAdapterIronSourceErrorInvalidServerParameters,
+                                                                          @"Interstitial 'appKey' parameter is missing. Make sure that appKey' server parameter is added.");
+        _interstitalAdLoadCompletionHandler(nil, error);
+        return;
+    }
+    
+    if (credentials[GADMAdapterIronSourceInstanceId]) {
+        self.instanceID = credentials[GADMAdapterIronSourceInstanceId];
+    } else {
+        [GADMAdapterIronSourceUtils onLog:@"Interstitial 'instanceID' parameter is missing. Using the default instanceID."];
+        self.instanceID = GADMIronSourceDefaultInstanceId;
+    }
+    
+    [[GADMediationAdapterIronSource alloc]
+     initIronSourceSDKWithAppKey:applicationKey
+     forAdUnits:[NSSet setWithObject:IS_INTERSTITIAL]];
+    
+    if (interstitialDelegate == nil) {
+        [GADMAdapterIronSourceUtils
+         onLog:[NSString
+                stringWithFormat:@"Interstitial adapterDelegate is null."]];
+        return;
+    }
+    if ([self canLoadInterstitialInstance]) {
+        [self setState:GADMAdapterIronSourceInstanceStateLocked];
+        [interstitialAdapterDelegates setObject:self forKey:self.instanceID];
+        [GADMAdapterIronSourceUtils
+         onLog:[NSString
+                stringWithFormat:@"Interstitial loadInterstitial for instance with ID: %@",
+                self.instanceID]];
+        [IronSource loadISDemandOnlyInterstitial:self.instanceID];
```

Comment:
@alon357 Does the ironSource SDK check internally whether it is initialized or not? If it isn't initialized at ad load time, will it attempt to get initialized?

Reply:
Hey @joshh-devrel, yes- the IronSource SDK checks this internally. An attempt to call Load before the SDK is initialized will fire LoadFailed : Cannot load before init. And no, it will not cause the SDK to attempt to initialize- init has to be explicitly called.